vendor/engines/saas/app/jobs/base_login_job.rb

```diff
@@ -0,0 +1,36 @@
+class LoginJob < ActiveJob::Base
+
+  def perform(params)
+    payload = map_payload(params)
+    Analytics::IdentifyUserJob.perform_later(payload)
+
+    #email = map_email(params)
+    #Queue job to send the email
+  end
+
+  def self.action(action)
+    @action = action
+  end
```

Comment:
Not 100% following you. This is the same pattern we have been using with the Analytics Jobs, and the Issue Jobs for Child Class instance vars.